tests/Unit/ApparentHorizons/Test_Strahlkorper.cpp

```diff
@@ -0,0 +1,426 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+#include <random>
+
+#include "ApparentHorizons/SpherepackIterator.hpp"
+#include "ApparentHorizons/Strahlkorper.hpp"
+#include "Numerical/RootFinding/QuadraticEquation.hpp"
+#include "tests/Unit/ApparentHorizons/YlmTestFunctions.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+
+void test_invert_spec_phys_transform() {
+  const double avg_radius = 1.0;
+  const double delta_radius = 0.1;
+  const size_t l_grid_init = 33;
+  const auto l_grid = static_cast<size_t>(l_grid_init * 1.5);
+  const std::array<double, 3> center = {{0.1, 0.2, 0.3}};
+
+  // Create radius as a function of angle
+  DataVector radius(YlmSpherepack::physical_size(l_grid_init, l_grid_init),
+                    avg_radius);
+  {
+    std::uniform_real_distribution<double> ran(0.0, 1.0);
+    std::mt19937 gen;
+    for (auto& r : radius) {
+      r += delta_radius * ran(gen);
+    }
+  }
+
+  // Initialize a strahlkorper of l_max=l_grid_init
+  const Strahlkorper<Frame::Inertial> strahl_initial(radius, l_grid_init,
+                                                     l_grid_init, center);
+
+  // Put that Strahlkorper onto a larger grid
+  const Strahlkorper<Frame::Inertial> strahl_final(l_grid, l_grid,
+                                                   strahl_initial);
+
+  // Compare coefficients
+  SpherepackIterator iter_init(strahl_initial.l_max(), strahl_initial.m_max());
+  SpherepackIterator iter_final(strahl_final.l_max(), strahl_final.m_max());
+  const auto& init_coefs = strahl_initial.coefficients();
+  const auto& final_coefs = strahl_final.coefficients();
+
+  for (size_t l = 0; l <= strahl_initial.ylm_spherepack().l_max(); ++l) {
+    for (int m = -static_cast<int>(l); m <= static_cast<int>(l); ++m) {
+      CHECK(init_coefs[iter_init.set(l, m)()] ==
+            approx(final_coefs[iter_final.set(l, m)()]));
+    }
+  }
+
+  for (size_t l = strahl_initial.ylm_spherepack().l_max() + 1;
+       l <= strahl_final.ylm_spherepack().l_max(); ++l) {
+    for (int m = -static_cast<int>(l); m <= static_cast<int>(l); ++m) {
+      CHECK(final_coefs[iter_final.set(l, m)()] == approx(0));
+    }
+  }
+}
+
+void test_avg_radius() {
```

Comment:
I'm not sure why this function exists...

It seems the `average_radius` testing would be better grouped with the radius and normal test. And it's unclear why copy/move semantics are tested in `test_avg_radius`.